cli/src/commands/extensions/mod.rs

```diff
@@ -100,14 +108,25 @@ pub async fn subcmd_list_extensions() -> CommandResult {
 }
 
 // Return a list of installed extensions. Filter out invalid extensions instead of exiting early.
-fn list_extensions() -> Result<Vec<Extension>> {
-    let extension_path = extensions_path()?;
+fn installed_extensions() -> Result<Vec<Extension>> {
+    let extensions_path = extensions_path()?;
 
-    if !extension_path.exists() {
+    if !extensions_path.exists() {
         return Ok(Vec::new());
     }
 
-    Ok(std::fs::read_dir(extension_path)?
-        .filter_map(|d| Extension::try_from(d.map(|d| d.path()).ok()?).ok())
+    Ok(std::fs::read_dir(extensions_path)?
+        .filter_map(|dir_entry| {
+            match dir_entry
+                .map_err(|e| anyhow!("{}", e))
```

Comment:
There's no need to make an ad-hoc error here. `std::io::Error` will cleanly convert to `anyhow::Error`

```suggestion
                .map_err(|e| e.into())
```